torch/_dynamo/eval_frame.py

```diff
@@ -820,8 +820,8 @@ def export(
     if pre_autograd:
         assert aten_graph, "pre_autograd=True can only be used when aten_graph=True"
     f = innermost_fn(f)
-    call_to_inspect = f.forward if isinstance(f, torch.nn.Module) else f
-    original_signature = inspect.signature(call_to_inspect)
+    f = f.forward if isinstance(f, torch.nn.Module) else f
```

Comment:
I have some concerns on f = f.foward when f is nn.Module. It seems that this approach may diverge from torch.compile. Could you please explain the reasoning behind this decision?

Reply:
When we switch to `torch.export`, the first thing it does is change the callable to `mod.forward`. So I'm just matching that behavior.

If we don't do this, `inspect` can't get the source file (it doesn't work on nn modules) so we cannot do the `skipfiles` check.

Reply:
Yeah, I got your point. But I'm still not very comfortable to change it in dynamo.export though as it could be a different entry point for some advanced users besides of torch.export.  Can we inspect(call_to_inspect) to get the skipfiles? I could stamp on it right away if we do it in this way :).

Reply:
Sure, let me try that.

> But I'm still not very comfortable to change it in dynamo.export though as it could be a different entry point for some advanced users besides of torch.export.

Regardless, I want to understand your concern better (so we can answer it in the future). I don't think we'd want "advanced users" to need direct access to `dynamo.export`: we need to have one public entry point, otherwise `torch.export` is not a good interface. Any information you would ever need should be returned from that entry point. 
* Maybe you are worried about guards, etc. in the output of `dynamo.export`?
* Maybe you are worried that `torch.export` is over-designed in its input?

@gmagogsfm

Reply:
I'm not worried about the API though and torch.export is the only public entry point. I think my concern is more from engineering/maintenance perspective: it's beneficial to avoid divergence behavior unless necessary e.g.: invariants such as nn.modules are wrapped with OptimizedModule and _torchdynamo_orig_callable points to the original forward method are preserved. This could make the system more maintainable in the long run.